hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -815,6 +816,15 @@ update env ledger st ev = case (st, ev) of
   (Open openState, NetworkEvent _ otherParty (AckSn snapshotSignature sn)) ->
     -- XXX: ttl == 0 not handled for AckSn
     onOpenNetworkAckSn env openState otherParty snapshotSignature sn
+  ( Open OpenState{headId = ourHeadId}
+    , OnChainEvent Observation{observedTx = OnDecrementTx{headId}}
```

Comment:
No need to change anything, but maybe let's keep a note for later: This is exactly the location where we should consider _what happens if this observed tx is rolled back_ here.

Reply:
Suggest to re-order these to follow the natural flow of things: Client input > ReqDec > OnDecrementTx